### PR TITLE
Add Linux member names for IPV4_HEADER

### LINKED ISSUE #2637: Wrong iphdr and tcphdr mapping
### Describe the bug

https://github.com/l3af-project/eBPF-Package-Repository/pull/54 is failing which tries to use iphdr and tcphdr in cross-platform code.

https://github.com/microsoft/ebpf-for-windows/blob/main/include/net/ip.h#L94 defines it as _IPV4_HEADER, but the field names in that struct don't match the expected field names in iphdr which is causing the windows test failures in https://github.com/l3af-project/eBPF-Package-Repository/pull/54.

### OS information

_No response_

### Steps taken to reproduce bug

Try to compile https://github.com/l3af-project/eBPF-Package-Repository/pull/54

### Expected behavior

Build should succeed

### Actual outcome

Build fails

### Additional details

```
D:/a/eBPF-Package-Repository/eBPF-Package-Repository/ratelimiting/ratelimiting_kern.c(103,27): error G1A4676F8: no member named 'ihl' in 'struct _IPV4_HEADER' [D:\a\eBPF-Package-Repository\eBPF-Package-Repository\build\ratelimiting\ratelimiting_kern.vcxproj]
      __u8 l4_offset = iph->ihl * 4; // ipv4 header length
                       ~~~  ^
D:/a/eBPF-Package-Repository/eBPF-Package-Repository/ratelimiting/ratelimiting_kern.c(114,31): error G1A4[67](https://github.com/l3af-project/eBPF-Package-Repository/actions/runs/5384061159/jobs/9771541350?pr=54#step:5:68)6F8: no member named 'doff' in 'struct tcphdr' [D:\a\eBPF-Package-Repository\eBPF-Package-Repository\build\ratelimiting\ratelimiting_kern.vcxproj]
      __u16 data_offset = tcph->doff * 4; // tcp header length
                          ~~~~  ^
```

https://github.com/microsoft/ebpf-for-windows/blob/main/include/net/tcp.h#L11 has `data_offset` instead of `doff`

### PR DESCRIPTION
**Pull request by hawkinsw**
## Description

Fix #2637 

## Testing

No new tests are added. 

## Documentation

N/A

### CODE REVIEW

**Review by Alan-Jowett — APPROVED** [review]

File: include/net/ip.h [hunk 1]
```diff
@@ -24,39 +24,72 @@ typedef struct _IPV4_HEADER
...
+    {
+        uint32_t saddr; // Linux field name.
+        uint32_t SourceAddress;
+    };
+    union
+    {
+        uint32_t daddr; // Linux field name.
+        uint32_t DestinationAddress;
+    };
 } IPV4_HEADER, *PIPV4_HEADER;
```
Comment (dthaler) [inline include/net/ip.h:93]: I believe you also need to add "iphdr" as another struct name

Comment (hawkinsw) [reply include/net/ip.h:93]: https://github.com/microsoft/ebpf-for-windows/blob/9186d691a3cc27c6fd5edf5c30c78abea36be0a1/include/net/ip.h#L127

Comment (hawkinsw) [reply include/net/ip.h:93]: I *think* this is where that is handled?

**Review by hawkinsw — COMMENTED** [review]
I think I addressed the comment!

**Review by dthaler — APPROVED** [review]

**Review by saxena-anurag — APPROVED** [review]

**Pull request merged**